model/imports.py

```diff
@@ -16,19 +16,19 @@ def import_model_from_spec(spec, det):
 
 class ImportModel(ABC):
     '''Abstract class for importation models'''
-    def detected(self, t):
-        pass
-
-    def undetected(self, t):
+    def __init__(self,
+                no_inf_compartments,
+                no_age_classes):
+        self.no_inf_compartments = no_inf_compartments
+        self.no_age_classes = no_age_classes
+        self.no_entries = no_inf_compartments * no_age_classes
+    def cases(self, t):     # Cases is a list of import functions
         pass
 
 
 class NoImportModel(ImportModel):
-    def detected(self, t):
-        return 0.0
-
-    def undetected(self, t):
-        return 0.0
+    def cases(self, t):
+        return zeros(self.no_entries,)
```

Comment:
Is that comma at the end necessary?

Reply:
I will defer to your Python knowledge here and say probably not.